### PR TITLE
PR: Improve PyDMImageView

### PR DESCRIPTION
**Pull request by fernandohds564**
Improve PyDMImageView adding pyqtproperties:
 - normalizeData: attribute to define if the colors are defined relative
to minimum and maximum of array or to a fixed scale;
 - width: user defined width, in case there is no EPICS info about it
(is overridden by widthChannel and widthFromImageChannel).
 - colorMapMin and colorMapMax: pyqt properties to set limits of colors
in colormap.
 - colormap: set the colormap to be used.
 - readingOrder: if the data should be read in Fortran or C style.

### CODE REVIEW

File: pydm/widgets/image.py [hunk 1]
```diff
@@ -153,23 +154,35 @@ def image_value_changed(self, new_image):
...
-                self._needs_reshape = True
-                # We'll wait to draw the image until we get the width.
-                return
-            self.image_waveform = new_image.reshape((int(self.image_width), -1), order='F')
         elif len(new_image.shape) == 2:
             self.image_waveform = new_image
-        self.data_max_int = np.amax(self.image_waveform)  # take the max value of the recieved image
-        self.setColorMap()  # to update the colormap immediately
+        elif self._width_from_image_channel:
+            self.imageWidth = new_image[0]
```
Comment (hhslepicka) [inline pydm/widgets/image.py]: @fernandohds564 is this required by some special IOC that you guys have? Which EPICS Module is producing this waveform where the first element is the width?

Comment (fernandohds564) [reply pydm/widgets/image.py]: @hhslepicka a while ago we talked to the guy responsible for developing the IOC which will provide beam images from the Slits in our transport lines and he said that possibly the Area Detector they were considering had this behavior. That's why we have implemented this option.
However I just talked to him and he said that there will be separate channels for the width of the array, so it seems this option is not really needed. If you think this option is not necessary I can remove it.

Comment (hhslepicka) [reply pydm/widgets/image.py]: @fernandohds564 yes, please! Keep the idea of the width property but remove this concept of `width_from_image_channel`.

Comment (fernandohds564) [reply pydm/widgets/image.py]: ok @hhslepicka !

Comment (fernandohds564) [reply pydm/widgets/image.py]: It is done, @hhslepicka !

**Comment by fernandohds564** [comment]
Hi @hhslepicka, I already implemented the changes you requested here too!

**Comment by hhslepicka** [comment]
Hi @fernandohds564, yes, I am still reviewing this one.

**Comment by fernandohds564** [comment]
Hi guys! Is there something wrong with this one? If there is something that should be handled differently or excluded, please, let me know. I could close it too, if you think it is not appropriate. No problem with that!

**Comment by hhslepicka** [comment]
Hi @fernandohds564,

This review is taking a bit longer than usual as we are fixing some
performance related issues at the PyDMImageView and we are going to
incorporate many of your fixes there.
I will keep you posted.

**Comment by fernandohds564** [comment]
Thanks @hhslepicka !

**Comment by fernandohds564** [comment]
@hhslepicka and @mattgibbs, this PR will not pass in Travis CI tests, because I use `Q_ENUMS`, which is not available in python2. I will close this PR.

**Pull request closed without merging**